models/src/main/java/com/vimeo/networking2/Album.kt

```diff
@@ -139,3 +145,24 @@ data class Album(
     override val identifier: String? = resourceKey
 
 }
+
+/**
+ * @see Album.layout
+ * @see AlbumLayoutType
+ */
+val Album.albumLayoutType: AlbumLayoutType
```

Comment:
I don't have a strong preference either way. The code is definitely boilerplate and annoying to write, but it's not worryingly error prone or difficult to maintain as the `get() =` is straight forward and the underlying logic is shared between every extension. Discoverability _might_ suffer with the usage of annotation processing for those reading source code. However, the only concern I have is in relation to the reference in the property documentation to the extension property. I'm not sure how well KDoc would do referencing a generated property. Regarding the feasibility of creating such an annotation processor, I would say that it's very easy. The design and implementation would look very similar to this simple example annotation processor I created for a talk: https://github.com/anthonycr/Tagger

@msya do you think it would be a good idea to generate these using an annotation processor, or would you prefer to keep them in the source of the file? If you prefer to generate them, I can ticket the work.